### PR TITLE
Initial implementation of jammy base builder

### PR DESCRIPTION
**Pull request by ForestEckhardt**
Adds support for all buildpacks that currently support jammy base
- Java
- Java Native Image
- Go
- Procfile

Checks some of the steps of #1 off

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: .github/.syncignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CODEOWNERS
+dependabot.yml
+workflows/push-image.yml
```
Comment (fg-j) [inline .github/.syncignore:3]:
```suggestion
```
There's now a parameterized push builder image workflow that will work for this repo:
https://github.com/paketo-buildpacks/github-config/commit/3b0bfa0fbf3366f54d7e1a0952a310c0371b308b

See the workflow in action here: https://github.com/paketo-buildpacks/builder-jammy-buildpackless-tiny/actions/runs/2666153033

Comment (ForestEckhardt) [reply .github/.syncignore:3]: Ok let me fix that up

File: README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Paketo Jammy Base Builder
+
+## `paketobuildpacks/builder-jammy-base`
+
+This builder uses the [Paketo Jammy Base
+Stack](https://github.com/paketo-buildpacks/jammy-base-stack) (Ubuntu Jammy
+Jellyfish build image, distroless-like run image) with buildpacks for Java,
```
Comment (fg-j) [inline README.md:7]:
```suggestion
Stack](https://github.com/paketo-buildpacks/jammy-base-stack) (Ubuntu Jammy
Jellyfish build and run images) with buildpacks for Java,
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+To see which versions of build and run images, buildpacks, and the lifecycle
+that are contained within a given builder version, see the
+[Releases](https://github.com/paketo-buildpacks/builder-jammy-base/releases) on this
+repository. This information is also available in the `builder.toml`.
+
+For more information about this builder and how to use it, visit the [Paketo
+builder documentation](https://paketo.io/docs/builders/).  To learn about the
+stack included in this builder, visit the [Paketo stack
+documentation](https://paketo.io/docs/stacks/) and the [Paketo Jammy Tiny Stack
+repo](https://github.com/paketo-buildpacks/jammy-base-stack).
```
Comment (fg-j) [inline README.md:19]:
```suggestion
documentation](https://paketo.io/docs/stacks/) and the [Paketo Jammy Base Stack
repo](https://github.com/paketo-buildpacks/jammy-base-stack).
```

File: builder.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+description = "Ubuntu jammy base image with buildpacks for Java, Go, and Procfile"
```
Comment (fg-j) [inline builder.toml:1]:
```suggestion
description = "Ubuntu 22.04 Jammy Jellyfish base image with buildpacks for Java, Go, and Procfile"
```
to match https://github.com/paketo-buildpacks/builder-jammy-buildpackless-base/blob/f995d3850c23600654f1da3621fb4089084f2b97/builder.toml#L1

File: .github/.syncignore (hunk 1 above)
Comment (ForestEckhardt) [reply .github/.syncignore]: Updated!

File: builder.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  [[order.group]]
+    id = "paketo-buildpacks/java"
+    version = "6.31.0"
+
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/go"
+    version = "1.7.0"
+
```
[the remark this answers is not in the document]
Comment (ForestEckhardt) [reply builder.toml]: Order should be correct now

File: smoke/java_test.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it("builds successfully", func() {
+			var err error
+			// this is OK, the app also works for running with a JVM
+			source, err = occam.Source(filepath.Join("testdata", "java-native-image"))
```
Comment (fg-j) [inline smoke/java_test.go:55]: Could we rename this test app to `java` for clarity? I think that'll make the comment unnecessary.

Comment (fg-j) [inline smoke/java_test.go:55]: I'm wondering if using the same app for both enables the smoke tests to pass with the flipped buildpack order in the `builder.toml`. Do the existing bionic base builder tests fail for java if you flip the detection order of java and java native image? If so,  I think that's useful behaviour to preserve in the tests.

Comment (ForestEckhardt) [reply smoke/java_test.go:55]: The test does fail when swapped

**Review by fg-j — APPROVED** [review]
LGTM 🙂

**Pull request merged**